src/proxy.rs

```diff
@@ -1208,7 +1295,6 @@ mod tests {
         drop(_lock);
 
         assert_eq!(baseline_proxies.contains_key("http"), false);
-        assert_eq!(disabled_proxies.contains_key("http"), false);
```

Comment:
`disabled_proxies` was testing the case where the previous `RegistryProxyValues` was of the form `(0, "http://...")` - in other words, the proxy had a value, but was disabled. Since we swapped just using `Option<String>`s for that bit of data, this case rolled into the previous `baseline_proxies` case where the platform proxies code returns us `None`.